portfolio/src/main/webapp/api-script/script.js

```diff
@@ -8,12 +8,47 @@ mapOptions = {
   }
 };
 
+google.charts.load('current', { 'packages': ['corechart'] });
+google.charts.setOnLoadCallback(loadChart);
 
-async function loadMap() {  
+async function loadChart() {
+  const response = await fetch("/nasa-data");
+  const payload = await response.json();
+
+  const data = new google.visualization.DataTable();
+  data.addColumn("string", "ID");
+  data.addColumn("number", "year");
+  data.addColumn("number", "budget");
+
+  // Per Google's documentation this column will affect how the color shifts, since
+  // I want a constant color for every bubble I'm including it but leaving it blank
+  data.addColumn("string", "color");
+  data.addColumn("number", "patents");
+
+
+  for (dataPoint of payload) {
+    data.addRow(["", dataPoint.year, dataPoint.budget, "", dataPoint.patents]);
```

Comment:
Does the charts API fill this in for you later? Otherwise, is there a reason to have an unused ID field?

Reply:
The ID field is unfilled because it automatically gets assigned as the label for the bubble. It's purely for aesthetic reasons. Following code is an example: https://developers.google.com/chart/interactive/docs/gallery/bubblechart#color-by-numbers